### PR TITLE
Fix/Loop Protect - Remove noprotect from saved code

### LINKED ISSUE #9114: Prevent infinite loops from recurring by removing //noprotect from saved code
One of the most common questions I get through email is "I used //noprotect and now my browser is stuck. What do I do?"

I get this email at least once every day. 

We should automatically remove //noprotect from their code before we load it into the browser, so that when they return to the challenge after crashing their browser, it won't crash their browser again

### PR DESCRIPTION
**Pull request by Bouncey**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/git-rebase#squashing-multiple-commits-into-one) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #9114 
#### Description

Removes `//noprotect` from saved code.

Replaces it with advice on if their browser crashed.

### CODE REVIEW

**Comment by Bouncey** [comment]
Is this more in line with the issue @BerkeleyTrue ?

File: client/commonFramework/init.js [hunk 1]
```diff
@@ -33,6 +33,12 @@ window.common = (function(global) {
 
   common.seed = common.arrayToNewLineString(common.challengeSeed);
 
+  common.replaceNoprotect = function replaceNoprotect(value) {
+    return value.replace(/noprotect/gi, 'If your browser has just crashed, ' +
```
Comment (BerkeleyTrue) [inline client/commonFramework/init.js]: It just needs to remove the noprotect. No need for the extra warning.

File: client/commonFramework/init.js [hunk 2]
```diff
@@ -33,6 +33,10 @@ window.common = (function(global) {
 
   common.seed = common.arrayToNewLineString(common.challengeSeed);
 
+  common.replaceNoprotect = function replaceNoprotect(value) {
+    return value.replace(/\/\/noprotect/gi, '');
```
Comment (BerkeleyTrue) [inline client/commonFramework/init.js]: 👍

**Comment by Bouncey** [comment]
Commit amended

**Comment by BerkeleyTrue** [comment]
@Bouncey can you test what would happen if I had this in my code

```js
//noprotect function() {alert('oh noooooeoss I shouldn't be running');}
```

**Comment by Bouncey** [comment]
I can see what it will do. It wont be commented out. Shall I amend so it removes the whole line?

**Comment by BerkeleyTrue** [comment]
I think the better solution is just to remove the word `noprotect` while leaving the comment

**Comment by Bouncey** [comment]
:+1:

**Comment by BerkeleyTrue** [comment]
You'll also need to add that function to this code here https://github.com/FreeCodeCamp/FreeCodeCamp/blob/87ef97f59e2a2f4dc425982f76f14a57d0900bcf/client/commonFramework/code-storage.js#L47

**Comment by Bouncey** [comment]
Do I need to insert the function after line 54?

**Comment by BerkeleyTrue** [comment]
Line 57 where it's saving the code to localStorage

File: client/commonFramework/code-storage.js [hunk 3]
```diff
@@ -54,6 +58,7 @@ window.common = (function(global) {
         console.log('unable to save to storage');
         return code;
       }
+      replaceNoprotect(code);
```
Comment (BerkeleyTrue) [inline client/commonFramework/code-storage.js]: You'll need to move this inline as strings are immutable this doesn't effect the line below

```js
localStorage.setItem(key + 'Val', replaceThing(code));
```

**Comment by Bouncey** [comment]
Forgot a semicolon

File: client/commonFramework/code-storage.js (hunk 3 above)
Comment (Bouncey) [inline client/commonFramework/code-storage.js]: Yea, that was a total guess. Ill sort it now though.

**Comment by benmcmahon100** [comment]
#9114

**Comment by BerkeleyTrue** [comment]
@Bouncey Thanks! Sorry it took so long to merge.

Happy Coding!

**Pull request merged**